src/net.c

```diff
@@ -1509,7 +1509,7 @@ coap_read_session(coap_context_t *ctx, coap_session_t *session, coap_tick_t now)
         session->last_rx_tx = now;
       }
       p = buf;
-      retry = bytes_read == (ssize_t)buf_len;
+      retry = (bytes_read == (ssize_t)buf_len);
       while (bytes_read > 0) {
```

Comment:
I am not convinced that this change fixes CID 1502427. That CID is most likely a false positive, though.

Reply:
Reverted change.  I'm inclined to agree with you.